### ISSUE TITLE
Invalid error response to bad OAuth token request 

### ISSUE THREAD

**Issue by pbuyle** [RobinQu/koa-oauth2orize#2]
When using the koa-oauth2orize's all-grant example, a bad request for an authorization request produce an `text/plain`` response with a 400 status.

```
$ curl -v --user "abc123:ssh-secret" -d 'grant_type=authorization_code' http://localhost:3000/oauth/token
* Hostname was NOT found in DNS cache
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 3000 (#0)
* Server auth using Basic with user 'abc123'
> POST /oauth/token HTTP/1.1
> Authorization: Basic YWJjMTIzOnNzaC1zZWNyZXQ=
> User-Agent: curl/7.35.0
> Host: localhost:3000
> Accept: */*
> Content-Length: 29
> Content-Type: application/x-www-form-urlencoded
> 
* upload completely sent off: 29 out of 29 bytes
< HTTP/1.1 400 Bad Request
< Content-Type: text/plain; charset=utf-8
< Content-Length: 11
< Date: Fri, 10 Oct 2014 23:50:19 GMT
< Connection: keep-alive
< 
* Connection #0 to host localhost left intact
Bad Request
```

According to the OAuth specifications, the expected response for an bad token request should have 400 status code and a `application/json` body with an error code . There is no issue when using oauth2orize's all-grant example (sample below). I guess something in the Koa bridge does not handle errors in a way that allow the oauth2orize error handling middleware to process the error and produce the expected response.

```
$ curl -v --user "abc123:ssh-secret" -d 'grant_type=authorization_code' http://localhost:4000/oauth/token
* Hostname was NOT found in DNS cache
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 4000 (#0)
* Server auth using Basic with user 'abc123'
> POST /oauth/token HTTP/1.1
> Authorization: Basic YWJjMTIzOnNzaC1zZWNyZXQ=
> User-Agent: curl/7.35.0
> Host: localhost:4000
> Accept: */*
> Content-Length: 29
> Content-Type: application/x-www-form-urlencoded
> 
* upload completely sent off: 29 out of 29 bytes
< HTTP/1.1 400 Bad Request
< X-Powered-By: Express
< Content-Type: application/json
< Set-Cookie: connect.sid=9fShixRhgsxjJmrx4UpleiRr.Fyy8huP%2BKOTptSMctwJHCUPFIUj2%2BajgOZI6V%2FUj4mw; path=/; expires=Sat, 11 Oct 2014 03:50:55 GMT; httpOnly
< Date: Fri, 10 Oct 2014 23:50:55 GMT
< Connection: keep-alive
< Transfer-Encoding: chunked
< 
* Connection #0 to host localhost left intact
{"error":"invalid_request","error_description":"missing code parameter"}
```

**Comment by RobinQu** [comment]
Thanks for head up. It seems error handler of oauth2orize is not correctly mounted. Will submit a fix ASAP.

**Comment by RobinQu** [comment]
Fix via 6f036a1e448989531cf5bcf58be4652f5f3276fb 

You should update your `oauth.js` accroding to this [diff](https://github.com/RobinQu/koa-oauth2orize/commit/dca844cdae6e93f0130e8cd89f7b0b491b7d078f#diff-bffa6875843a7f652fff55ca3f34a8c9R216)
